app/services/order_detail_update_param_hash_extractor.rb

```diff
@@ -0,0 +1,17 @@
+class OrderDetailUpdateParamHashExtractor
+  attr_reader :params
+
+  def initialize(params)
+    @params = params
+  end
+
+  def updates_as_hash
```

Comment:
This method is really difficult to reason about, but I'm not sure how to make it clearer. I think a comment explaining what's it's doing would help. Also, the `updates` part of the method name implies it's going to be actually doing something.  I think maybe something like `extract` or even just `to_h` might be clearer.